src/components/Web3Button.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect } from 'react';
```

Comment:
mixed use of direct imports for `useEffects` and indirect `React.useState`, should stick to one pattern.